### PR TITLE
Fix _apply_verdict UnboundLocalError: bind cooldown from verdict (#48)

### LINKED ISSUE #48: _apply_verdict raises UnboundLocalError on every coollable failure — no cooldown or breaker streak has been applied since #42
## Symptom

Every coollable failure logged by the live gateway produces:

```
LiteLLM:ERROR: litellm_logging.py:3648 - LiteLLM.LoggingError: [Non-Blocking]
Exception occurred while failure logging cannot access local variable 'cooldown'
...
UnboundLocalError: cannot access local variable 'cooldown' where it is not
associated with a value
```

with the traceback ending at `switchyard/hooks.py` in `_apply_verdict`. This is firing constantly in production today (07:54:45, 07:55:23, 07:55:33 … in the current gateway log).

## Root cause

`switchyard/hooks.py` (`_apply_verdict`), introduced by 05d4aa2 (#36/#42):

```python
        did_cool_atomic = False
        if verdict.outcome is Outcome.TRANSIENT:
            breaker = self.registry.settings.transient_breaker
            if breaker.enabled:
                streak = await self.slots.bump_and_cool(
                    plan.key,
                    base=cooldown,          # ← never assigned
                    cap=breaker.max_seconds,
                    reason=verdict.outcome.value,
                )
                cooldown = escalated_cooldown(cooldown, streak, ...)
```

and the fallback path:

```python
        if verdict.should_cool and not did_cool_atomic:
            await self.slots.cool_down(plan.key, cooldown, verdict.outcome.value)
```

`cooldown` is **never assigned anywhere in the function**. The pre-#42 code cooled with `verdict.cooldown_seconds`; the refactor renamed the use sites but dropped the binding. So the function raises `UnboundLocalError` for **every verdict with `should_cool`** — RATE_LIMITED, TRANSIENT, QUOTA_EXHAUSTED, PLAN_DEAD, AUTH. Only CONTEXT/BAD_REQUEST (`is_our_fault` early return) survive.

Litellm swallows the exception as a non-blocking callback error, so requests complete — but since the rebuild:

- **no plan has ever been cooled down** (quota walls, rate limits, 5xx — nothing sticks),
- the transient-failure breaker never bumps its streak (`bump_and_cool` never runs),
- sessions are never dropped from quota-dead plans (`drop_lease` is after the crash).

Ledger failure counts and `note_concurrency_rejection`/learner notes still land because they run before the cooling section — which makes the damage quiet: usage looks right, cooldown state is just silently absent.

This directly fed the lane-spill incident in #47: a full `opencode-go` should have been cooled out of `judge` after its first capacity 429; instead it stayed at the front of the lane for every retry.

## Why the tests missed it

The #42 tests pin `SlotTable.bump_and_cool` (the Lua script) directly. Nothing drives `_apply_verdict` end-to-end with a coollable verdict — `grep -rn _apply_verdict tests/` finds nothing — and the suite's Redis/fake-redis fakes don't execute this path.

## Fix

1. Bind the verdict's cooldown once, before the cooling section:

   ```python
   cooldown = verdict.cooldown_seconds
   ```

   (or use `verdict.cooldown_seconds` at both use sites).
2. Add a `_apply_verdict` unit test per outcome class — at minimum TRANSIENT with the breaker enabled and RATE_LIMITED — asserting the cooldown key lands in Redis and no exception escapes. Check the test count goes up when appending (see CLAUDE.md's note on the `__main__` runner).

Related: #47 (same failure path; the crash removed the cooldown that would have kept a full plan out of its lane).

### PR DESCRIPTION
**Pull request by Fledgewing**
## What broke

The #42 refactor (`Retry once on transient upstream failures; escalate repeated cooldowns`) renamed the cooldown use sites in `_apply_verdict` but dropped the `cooldown = verdict.cooldown_seconds` binding. After that landed, **every** coollable verdict (RATE_LIMITED, TRANSIENT, QUOTA_EXHAUSTED, PLAN_DEAD, AUTH) raised `UnboundLocalError: cannot access local variable 'cooldown'` the first time it tried to cool a plan.

The damage was quiet because the ledger counts (`note_concurrency_rejection`, `note_reported`) still landed *before* the cooling section crashed — so the visible symptom was missing cooldowns, not stack traces. In particular:

- The **TRANSIENT** path never called `bump_and_cool`, so the transient-failure streak was never bumped and the escalated ladder never fired.
- **RATE_LIMITED** / **QUOTA_EXHAUSTED** / **PLAN_DEAD** / **AUTH** paths never wrote their `sy:cool:` key.
- **QUOTA_EXHAUSTED** / **PLAN_DEAD** / **AUTH** never called `drop_lease`, so a session stayed pinned to dead capacity instead of re-leasing onto a peer.

This is the cooldown gap behind the lane-spill incident in #47: traffic kept re-feeding dead or rate-limited plans because nothing ever told the picker to back off.

## The one-line fix

`switchyard/hooks.py` — bind `cooldown = verdict.cooldown_seconds` immediately before `did_cool_atomic = False` (line 661). This mirrors pre-#42 behaviour and feeds both branches of the cooling section:

- The `TRANSIENT` branch reads it as `base=cooldown` for `bump_and_cool`, then recomputes via `escalated_cooldown(cooldown, streak, cap=...)`. The atomic INCR/EXPIRE Lua already used `cooldown` correctly — the binding is what gave it a value to start from.
- The `cool_down` branch (`if verdict.should_cool and not did_cool_atomic`) reads it directly. The `drop_lease` branch is downstream of the binding and was never missing the variable — it just never ran because the cooling section crashed first.

Why that exact binding point: pre-#42 the binding lived at the top of the "Apply the cooldown" comment block, and the `TRANSIENT` ladder was preserved via `escalated_cooldown`/`bump_and_cool`. The fix restores that position verbatim.

## Tests

`tests/test_verdict.py` — new file, five FakeRedis end-to-end tests driving `_apply_verdict` directly (no Picker, no provider, no real Redis):

1. `test_transient_with_breaker_writes_cooldown_key_and_respects_cap` — eight consecutive TRANSIENT failures; the streak grows 1..8 and the cooldown lands at the breaker cap (1800s) once `60 × 2^(n-1)` exceeds it. **Regression proof**: with the `cooldown = ...` line stashed, the first call raises the original `UnboundLocalError`.
2. `test_transient_without_breaker_uses_cool_down_with_verdict_cooldown` — breaker disabled, `bump_and_cool` is skipped, the verdict's own cooldown lands via `cool_down` and the streak stays at 0.
3. `test_rate_limited_writes_cooldown_key_and_raises_no_exception` — RATE_LIMITED writes the `sy:cool:` key at the verdict's value; the test explicitly asserts no exception escapes (the Litellm callback would otherwise swallow a bug here).
4. `test_quota_exhausted_writes_cooldown_and_drops_lease` — QUOTA_EXHAUSTED both cools the plan **and** drops the session lease, so the next pick re-leases onto a peer rather than re-feeding dead capacity.
5. `test_double_count_guard_skips_second_call_with_same_request_id` — same `ctx.request_id` short-circuits the second `_apply_verdict`; a fresh `request_id` is allowed through. Without the per-attempt marker, the streak would double and the cooldown would jump straight to the cap on a single 5xx.

## Verification

Offline suite, all green from this worktree:

```
$ python3 tests/test_verdict.py
... 5 verdict tests passed

$ python3 tests/test_routing.py tests/test_classify.py 2>/dev/null
... all routing tests passed

$ for t in tests/test_*.py; do python3 "$t" >/dev/null || echo "FAIL $t"; done
(no FAILs)

$ python3 tests/render_preview.py
rendered index.html + fragments (87900 bytes) -> /tmp/switchyard-preview.html
  10 plans, 19 models, 5 lanes, 18 quota windows, 4 probes
```

## Deploy note

`switchyard/` is `COPY`ed into the gateway / portal / sidecar images — only `./config` is mounted from the host. This commit is **inert until the main checkout rebuilds and redeploys**:

```bash
# from the main checkout, not from this worktree
docker compose build gateway portal claude-max-sidecar xai-token-proxy
docker compose up -d
```

Closes #48.

### CODE REVIEW

File: tests/test_verdict.py [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+# someone's real config would otherwise silently become the fixture.
+os.environ["SWITCHYARD_PLANS"] = plans_path()
+
+from dataclasses import replace  # noqa: E402
+
+from switchyard import models  # noqa: E402
+from switchyard.classify import Outcome, Verdict  # noqa: E402
+from switchyard.hooks import SwitchyardHandler  # noqa: E402
+from switchyard.policy import CapacityPolicy  # noqa: E402
+from switchyard.slots import K_COOL, K_LEASE  # noqa: E402
```
Comment (Fledgewing) [inline tests/test_verdict.py]: **[nit] Unused imports K_COOL / K_LEASE**

Neither name is referenced anywhere in the test file — `K_COOL` only shows up in a comment (line 234) and `K_LEASE` doesn't appear at all. The cooldown and lease shapes are already covered transitively by `slots.cooldown_state(...)` and `slots.get_lease(...)`, so the explicit constants add noise without sharpening any assertion. Other tests in the suite (e.g. `test_routing.py`) match this convention and don't import them either. Drop both names from the import line.

**Comment by Fledgewing** [comment]
Reviewed 1 finding: 0 blocker(s), 0 should-fix, 1 nit — see inline comment at 3ab517d.
